Added details of the OSU SubCritical Assembly and code specifics. (#3)
scram-lab/OSU-subcriticalassembly: README.md

# OSU Subcritical Assembly

This repository contains an OpenMC model of the subcritical assembly at Oregon State University's Radiation Center. The assembly is composed of graphite bricks, UO_2 fuel rods, a concrete reflector, and a plutonium–beryllium neutron source.

## Overview

The subcritical assembly is designed to teach students about nuclear reactions and how to take measurements for a reactor. The main components of the assebmbly are as follows:

- **Graphite Bricks:** Serve as both structural support and neutron moderator.
- **UO_2 Fuel:** Annular fuel rods with natural uranium enrichment (0.072% U-235) and aluminum cladding.

> - **Concrete Reflector:** Enhances neutron economy by reflecting neutrons back into the assembly.

Reviewer: "Reflector" will likely be interpreted by readers as surrounding the reactor core. Perhaps you can say "concrete structural material" and specify that this is the floor and wall next to the assembly, and state that it is also a reflector.

Author: Understood and edited to reflect as such.

- **Plutonium–Beryllium Neutron Source:** Initiates the neutron population required for the assembly's operation.

## Assembly Dimensions

### UO₂ Fuel Rods
- **Type:** Annular rods with aluminum cladding.
- **Enrichment:** Natural uranium (0.072% U-235).
- **Dimensions:** 
  - **Fuel Length in z-direction:** 45 cm
  - **Innermost Air Hole Radii:** 0.46 cm
  - **Clad Inner Radii:** 0.61875 cm
  - **Radii of the UO_2 Fuel:** 1.76125 cm

>   - **Radii of Outermost Clad:** 1.92 cm

Reviewer: This is great. I know you have the locations of the centerpoints of the fuel rods, but do you have a notion of the pitch that separates them?

Author: Yes, I have updated the README to add that detail; the pitch is 15 cm.

### Graphite Bricks
- **Dimensions:**
  - **Length in x-direction:** 50 cm
  - **Height in y-direction:** 15 cm
  - **Width in z-direction:** 15 cm
- **Fuel Arrangement:**  

>   Each brick features a central hole for fuel placement that is the same radii as the outermost cladding of the fuel. Along the z-axis of the brick, three UO_2 fuel rods are inserted. The combined length of these rods is 135 cm, with an additional 15 cm gap filled by air.

Reviewer: Say that the rods are inserted "end to end" so it is clear that they are in a single hole and not three separate holes in a concrete block. 

This description of the brick arrangement I'm having trouble visualizing. are there 10 or so bricks stacked in the z-direction to make a 150cm long hole for the rods?

Author: Noted and edited. 

I have reworded the description and hopefully described the arragement better.

### Concrete Reflector
- **Dimensions:**
  - **Length in x-direction:** 15.24 cm
  - **Height in y-direction:** 180 cm
  - **Width in z-direction:** 150 cm
- **Placement:**  
  Positioned to the right of the assembly, the concrete wall functions as a reflector to bounce neutrons back into the reactor core. This concrete reflector's leftmost side is touching the rightmost side of the assembly.

### Graphite Floor
- **Dimensions:**
  - **Length in x-direction:** 150 cm
  - **Height in y-direction:** 30 cm
  - **Width in z-direction:** 150 cm

### Neutron Source
- **Type:** Plutonium–Beryllium.
- **Location:**  
  Centrally located on the graphite floor at coordinates (75, -15, 75) with a raii of 2 cm within the reactor framework.

> 

Reviewer: I know that we don't have explicit references for this model, but I think it would be good to say where you got the information to construct it. Perhaps "we would like to thank Dr. Reese for providing information on the locations of fuel elements used to construct this model".

Author: Added the acknowledgment.

# OSU Subcritical Assembly

This repository contains an OpenMC model of the subcritical assembly at Oregon State University's Radiation Center. The assembly is composed of graphite bricks, UO_2 fuel rods, a concrete reflector, and a plutonium–beryllium neutron source.

## Running the Python Script

> - Running the RadCenterSubcriticalAssemblyOUS.ipynb will require OpenMC to be installed correctly on your computer. When running the script, look at the model.run command at the very end of the script; currently, it is being multithredded to 20 threads. Change this number to what ever you feel is necessary as long as you have enough threads in your CPU. To see how many CPU threads you have, look at Task Manager and click on the Performance Tab, there you will see on the bottom-left a list of technical specifications on your CPU, look to Logical Processors, that will be the number of threads you have.

Reviewer 2: ```suggestion
- Running the RadCenterSubcriticalAssemblyOSU.ipynb will require OpenMC to be installed correctly on your computer. When running the script, look at the model.run command at the very end of the script; currently, it is being multithreaded to 20 threads. Change this number to whatever you feel is necessary as long as you have enough threads in your CPU. To see how many CPU threads you have, look at Task Manager and click on the Performance Tab, there you will see on the bottom-left a list of technical specifications on your CPU, look to Logical Processors, that will be the number of threads you have.
```

- Take note as well to settings.particles and mesh.dimension, these values can be adjusted by the user as needed. Currently, the code will be very demanding on 20 cores due to the high values and will take a long time to compile.

> - To run any analysis on the Analysis.ipynb script, make sure the outputs from the RadCenterSubcriticalAssemblyOUS.ipynb such as the tallies.xml, summary.xml, and the statepoint file are placed in the analysis folder. The script will not run at all if they are not present.

Reviewer 2: ```suggestion
- To run any analysis on the Analysis.ipynb script, make sure the outputs from the RadCenterSubcriticalAssemblyOSU.ipynb such as the tallies.xml, summary.xml, and the statepoint file are placed in the analysis folder. The script will not run at all if they are not present.
```

> - If you change any parameters on the RadCenterSubcriticalAssemblyOUS.ipynb such as mesh dimension size, or the number of batches, some parameters on the Analysis.ipynb will need to be changed as a result to match. Running the script will tell you what you need to change. 

Reviewer 2: ```suggestion
- If you change any parameters on the RadCenterSubcriticalAssemblyOSU.ipynb such as mesh dimension size, or the number of batches, some parameters on the Analysis.ipynb will need to be changed as a result to match. Running the script will tell you what you need to change. 
```
